### PR TITLE
Introduce an 'svc' segment for DNS search

### PR DESCRIPTION
**Pull request by thockin**
@brendandburns @vishh @smarterclayton @dchen1107

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -864,8 +864,12 @@ func (kl *Kubelet) getClusterDNS(pod *api.Pod) ([]string, []string, error) {
 		dns = append([]string{kl.clusterDNS.String()}, hostDNS...)
 	}
 	if kl.clusterDomain != "" {
-		nsDomain := fmt.Sprintf("%s.%s", pod.Namespace, kl.clusterDomain)
-		dnsSearch = append([]string{nsDomain, kl.clusterDomain}, hostSearch...)
+		// TODO(vishh): Remove the oldNsDomain entry once the DNS crossover to inject
+		// "svc" is done.
+		oldNsDomain := fmt.Sprintf("%s.%s", pod.Namespace, kl.clusterDomain)
+		nsSvcDomain := fmt.Sprintf("%s.svc.%s", pod.Namespace, kl.clusterDomain)
+		svcDomain := fmt.Sprintf("svc.%s", pod.Namespace, kl.clusterDomain)
```
Comment (smarterclayton) [inline pkg/kubelet/kubelet.go]: Typo?  Two format Args but only one replacement

**Comment by dchen1107** [comment]
LGTM.

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (thockin) [inline pkg/kubelet/kubelet.go]: dammit., fixed

Comment (dchen1107) [inline pkg/kubelet/kubelet.go]: ahh, you are right!

**Comment by dchen1107** [comment]
LGTM again!

**Comment by thockin** [comment]
I need this to go in before 8:00 to make the next release cut.

**Comment by smarterclayton** [comment]
LGTM

**Comment by nikhiljindal** [comment]
Shippable is green. Merging.

**Comment by dchen1107** [comment]
Thanks!

**Pull request merged**